### PR TITLE
Various fixes to pages in references flow (part 2)

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
## Context

Bugs fixes and tidy up, following on from #3341.

### Fixes
* Remove deprecated reference request start page
* Ensure page heading always sits inside form body and below any error summary
* Remove errant `govuk-body` class on a link

### Clean up
* Consistent multi-line `form_with()` for reference forms
* Consistent whitespace between elements in reference forms
* Consistent HTML syntax in reference forms
* Add missing reference localisations
* Use `f.govuk_submit` helper

## Guidance to review

This PR removes the following route, controller and view:

`get '/request/:id/start' => 'references/request#start', as: :references_start_request`

This seems to now be redundant (and the `TODO` message in the controller would seem to confirm this suspicion).

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by frankieroberto — APPROVED** [review]
Looks good.

Seems like you've added a trailing comma to some of the multiline `form_with()` method calls, but not for all of them. Is there a preferred style for this?

File: config/locales/application_form.yml [hunk 1]
```diff
@@ -408,16 +408,23 @@ en:
...
-          character: 'For example: a mentor, or someone you know from volunteering.'
+        academic:
+          label: Academic
+          hint_text: 'For example: a university tutor.'
+        professional:
+          label: Professional
+          hint_text: 'For example: a manager.'
+        school_based:
+          label: School-based
+          hint_text: 'For example: a colleague from a school where you did experience.'
```
Comment (stevehook) [inline config/locales/application_form.yml:419]: _did experience_ sounds odd to me but maybe it works in context.

**Review by stevehook — APPROVED** [review]
All looks fine to me. I think we devs probably do need a reminder about conventions etc.

**Comment by stevehook** [comment]
> Looks good.
> 
> Seems like you've added a trailing comma to some of the multiline `form_with()` method calls, but not for all of them. Is there a preferred style for this?

It is the preferred style in Ruby method calls that span multiple lines. In plain ruby files its enforced by Rubocop but not in `.erb` files.

**Comment by paulrobertlloyd** [comment]
@frankieroberto 

> Seems like you've added a trailing comma to some of the multiline `form_with()` method calls, but not for all of them. Is there a preferred style for this?

We don’t lint ERB templates (yet?), but when I’ve made changes to other Ruby files, rubocop complains if last items honest have a trailing comma, so am assuming that’s the preferred style.

**Pull request merged**